### PR TITLE
Replace Travis with GitHub actions

### PR DESCRIPTION
**Pull request by mlevesquedion**
This PR replaces Travis CI with GitHub actions.

This change is prompted by the migration from `travis-ci.org` to `travis-ci.com`. 

A few advantages of this transition:
- Jobs start faster and run faster
- Timing information for each check is provided

I have added the `yamllint` check, but left it commented out for now. Indeed, `yamllint` is currently failing on `master`. This will be remedied by this follow-up PR: #269.

Note that the diff shows an existing GitHub Actions workflow being modified, instead of a brand-new workflow. This is because earlier on I added the sample workflow from GitHub to get started with a basic example.

- (N/A) [ ] Running against a large codebase such as [Kubernetes](https://github.com/kubernetes/kubernetes) does not error out. (See [DEVELOPING.md](https://github.com/google/go-flow-levee/blob/master/DEVELOPING.md) for instructions on how to do that.)
- (N/A) [ ] Appropriate changes to README are included in PR

### CODE REVIEW

**Review by vinayakankugoyal — APPROVED** [review]

**Review by PurelyApplied — COMMENTED** [review]
I think it might be nice if we separate our actual-tests `checks` and the other linter checks.  There's always a touch of affront when I see the failure `X` when I'm sure I just ran tests.

I'm new to GitHub Actions, so it's not clear to me what is containerized and what isn't.  For instance, it seems strange to me that installing `goimports` doesn't impact the `go mod` or `git diff` checks.  If these `steps` are order dependent, it might make sense to split them into separate `checks`.

**Comment by mlevesquedion** [comment]
> I think it might be nice if we separate our actual-tests checks and the other linter checks.

Sure, I can do that.

> I'm new to GitHub Actions, so it's not clear to me what is containerized and what isn't. For instance, it seems strange to me that installing goimports doesn't impact the go mod or git diff checks. If these steps are order dependent, it might make sense to split them into separate checks.

I'm also new to GitHub Actions. AFAIU everything runs in order, in the same environment. I think the reason that **installing** `goimports` is fine is that this merely installs a binary in `$GOBIN`, it doesn't touch the files in the repo. When running `goimports`, since we're using the `-d` flag, the files in the repo won't be touched by that either, it just prints the diff. If there is a diff, the goimports check will fail and none of the subsequent checks will run.

**Review by PurelyApplied — CHANGES_REQUESTED** [review]
I think I was crossing wires with the behavior of `GO111MODULE=on go install ...`.  That installes it to the local project, which updates `go.mod`.  So 👍 there.

Is test failure based on exit value?  I note that `goimports` returns `0` regardless of whether or not there's a diff.  That's why the Travis explicitly called `false` to return nonzero.  Ditto for `git diff` without the `--quiet` option.  From the Travis, it looks like `staticcheck` should return nonzero as expected.  I'm unfamiliar with `addlicense`, so please investigate that.

**Comment by PurelyApplied** [comment]
Oh!  I was unclear.  I had really just meant splitting out all the lint steps into one other job, and the go tests as a second job.  Something like

```yaml
jobs:
  tests:
    name: Go tests
    #...
  lint-checks:
    steps:
      - name: License check
    # ...
      - name: goimport check
    # ...
# ...
```

But expanding them out is fine by me, too.  Saves you the click if you hit one of them - you already know what's wrong and what you need to go fix.  (For me, it's going to be forgetting licenses on new files.)

**Comment by mlevesquedion** [comment]
> Is test failure based on exit value?

Yes.

> I note that goimports returns 0 regardless of whether or not there's a diff.

Oops. Bad assumption by me here. Should be fixed now. Same for git diff.

> ... addlicense ... 

I did investigate addlicense. It returns non-zero when a license is missing.

> Job/Check splitting

Gotcha, no worries. The main downsides I see to using jobs instead of steps is that 1. it's slightly slower overall, 2. the first two steps of each job are duplicated. But it's nice to have more detailed information on the PR itself, without having to click through.

**Comment by mlevesquedion** [comment]
I did some experimenting and I think the best way to go is to have a single `CI` workflow with separate jobs for `lint` and `test`, as you originally suggested.

LMK if you think something else would work better.

**Review by PurelyApplied — COMMENTED** [review]
> It looks like GitHub Actions doesn't support YAML anchors

👀 
That's... like... core to YAML.  Although maybe it's an exposure to recursive expansion or something that could blow up their CI.  \shrug.

> Job/Check splitting

No strong preference.  Looks good as is.

----

It looks like failing a single step terminates the job.  See #270 / [the associated test run](https://github.com/google/go-flow-levee/runs/1793839162).  So if you fail an early lint check, the later ones don't get executed.  There is a `continue_on_error` key we could add, but that would let failing jobs pass unless we started doing something funky with flag files.

I think the frequency with which we hit linter issues is low enough that this isn't really an issue.  While this could push us towards splitting out linter checks into their own jobs like before, I like the cleanliness of a single linter checkbox.

**Review by PurelyApplied — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+env:
+  GO_VERSION: "1.14"
```
[the remark this answers is not in the document]
Comment (benhxy) [reply .github/workflows/ci.yaml:14]: +1. setup-go is version agnostic. It just downloads whichever version you specify.

**Review by benhxy — APPROVED** [review]
LGTM

File: .github/workflows/ci.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (mlevesquedion) [reply .github/workflows/ci.yaml:14]: GHA certainly supports 1.15, and likely so do all of our checks. I was just using the same version we have in `go.mod`, as well as the version we were using in Travis.

If you want to bump to 1.15, that would be a good future PR.

**Pull request merged**